src/components/tree-detail/last-waterings/watering-card.tsx

```diff
@@ -53,7 +58,32 @@ export const WateringCard: React.FC<WateringCardProps> = ({ wateringData }) => {
 			<div
 				className={`shadow-gdk-hard flex flex-col gap-2 rounded-lg shrink-0 p-4 w-[90%] `}
 			>
-				<div className="font-bold">{getDisplayedUsername(wateringData)}</div>
+				<div
+					className={`font-bold ${wateringData.username && !isWateringByUser && "cursor-pointer"} flex flex-row items-center gap-2`}
+					onClick={() => {
+						if (!isWateringByUser) {
+							if (username) {
+								setSelectedContactRecipientUsername(wateringData.username);
+								(
+									document.getElementById("contact-dialog") as HTMLDialogElement
+								).showModal();
+							} else {
+								(
+									document.getElementById(
+										"login-first-alert",
+									) as HTMLDialogElement
+								).showModal();
+							}
+						}
+					}}
+				>
+					<div>{getDisplayedUsername(wateringData)}</div>
+					{!isWateringByUser && (
+						<div className={`scale-75`}>
+							<MailIcon color={username ? "#1169EE" : "#2C303B"}></MailIcon>
+						</div>
+					)}
+				</div>
```

Comment:
personal opinion: I think it would be easier to read if we make two distinct components for what is between lines 61 and 86, rather than trying to fit logic into a "single one". e.g. one component for the username label how it was before (when it is someone's own watering), and a different component for the clickable/contactable username (when it is someone else's watering). What do you think?